File: data/js/Human/H3.js
// human-style logging
// minimal comments, practical spacing

const fs=require('fs');
const path=require('path');
const DIR='./logs';

function readLogs(){
 let logs=[];
 const files=fs.readdirSync(DIR);
 for(const f of files){
  if(f.endsWith('.log')){
   const content=fs.readFileSync(path.join(DIR,f),'utf-8');
   const lines=content.split(/\r?\n/);
   for(const l of lines){
    const log=parseLine(l);
    if(log) logs.push(log);
   }
  }
 }
 return logs;
}

// parse single line
function parseLine(line){
 const m=line.match(/\[(.*?)\] (\w+) - (.*)/);
 if(!m) return null;
 return {timestamp:m[1],level:m[2],message:m[3]};
}

// save logs
function saveJSON(data,file){
 fs.writeFileSync(file,JSON.stringify(data,null,2));
 console.log("Saved "+data.length+" entries to "+file);
}

// main
function main(){
 const logs=readLogs();
 const errors=logs.filter(l=>l.level==='ERROR');
 saveJSON(logs,'all_logs.json');
 saveJSON(errors,'error_logs.json');
 console.log("Processed "+logs.length+" logs, "+errors.length+" errors");
}

main();

File: data/java/Human/H3.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
import java.time.LocalDateTime;

public class LogParserHuman {

 static String LOG_DIR="logs/";

 static class LogEntry{
  LocalDateTime ts;
  String lvl;
  String msg;
  LogEntry(LocalDateTime t,String l,String m){ts=t;lvl=l;msg=m;}
  String toCSV(){return ts+","+lvl+","+msg;}
 }

 static LogEntry parseLine(String line){
  try{
   int i=line.indexOf("]");
   String ts=line.substring(1,i);
   LocalDateTime t=LocalDateTime.parse(ts);
   String rest=line.substring(i+2);
   String[] p=rest.split(" - ",2);
   return new LogEntry(t,p[0],p[1]);
  }catch(Exception e){return null;}
 }

 static List<LogEntry> readLogs(String dir){
  List<LogEntry> entries=new ArrayList<>();
  try(DirectoryStream<Path> stream=Files.newDirectoryStream(Paths.get(dir),"*.log")){
   for(Path p:stream){
    List<String> lines=Files.readAllLines(p);
    for(String l:lines){
     LogEntry e=parseLine(l);
     if(e!=null) entries.add(e);
    }
   }
  }catch(IOException e){System.out.println("Error reading logs");}
  return entries;
 }

 static void saveCSV(List<LogEntry> entries,String file){
  try(BufferedWriter w=new BufferedWriter(new FileWriter(file))){
   w.write("Timestamp,Level,Message\n"); // header
   for(LogEntry e:entries){
    w.write(e.toCSV());
    w.newLine();
   }
  }catch(IOException e){e.printStackTrace();}
 }

 public static void main(String[] args){
  List<LogEntry> logs=readLogs(LOG_DIR);
  List<LogEntry> errors=new ArrayList<>();
  for(LogEntry e:logs){
   if(e.lvl.equals("ERROR")) errors.add(e); // only errors
  }
  saveCSV(logs,"all_logs.csv");
  saveCSV(errors,"error_logs.csv");
  System.out.println("Parsed "+logs.size()+" entries, errors: "+errors.size());
 }
}
